cpp/src/io/json/write_json.cu

```diff
@@ -247,6 +251,41 @@ std::unique_ptr<column> struct_to_strings(table_view const& strings_columns,
                    thrust::make_counting_iterator<size_type>(0),
                    thrust::make_counting_iterator<size_type>(total_rows),
                    scatter_fn);
+  if (!include_nulls) {
+    // if previous column was null, then we skip the value separator
+    rmm::device_uvector<bool> d_str_seperator(total_rows, stream);
+    auto row_num = cudf::detail::make_counting_transform_iterator(
+      0, [tbl = *tbl_device_view] __device__(auto idx) -> size_type {
+        return idx / tbl.num_columns();
+      });
+    auto validity_iterator =
+      cudf::detail::make_counting_transform_iterator(0, validity_fn{*tbl_device_view});
+    thrust::exclusive_scan_by_key(rmm::exec_policy(stream),
+                                  row_num,
+                                  row_num + total_rows,
+                                  validity_iterator,
+                                  d_str_seperator.begin(),
+                                  false,
+                                  thrust::equal_to<size_type>{},
+                                  thrust::logical_or<bool>{});
+    thrust::for_each(rmm::exec_policy(stream),
```

Comment:
I've been trying to find a way to transform this into a `thrust::copy_if()`, to avoid the `for_each()`. I don't think it's necessarily cleaner.

`for_each()` FTW.